lib/watir_drops/page_object.rb

```diff
@@ -18,52 +20,68 @@ def page_url
         end
       end
 
+      def page_title
+        define_method("page_title") do |args|
+          yield(*args)
+        end
+      end
+
       def element_list
         @element_list ||= []
       end
 
+      def required_element_list
+        @required_element_list ||= []
+      end
+
       def inherited(subclass)
         subclass.element_list = element_list.dup
+        subclass.required_element_list = required_element_list.dup
       end
 
-      def elements(name, &block)
+      def elements(name, required = false, &block)
```

Comment:
Just made some updates. Added some more tests. Added rescue to on_page to let rspec catch the exceptions. In Visit, I'm raising Waitr::Exception::Error Now. Still not very descriptive, any suggestion?